lib/book.js

```diff
@@ -0,0 +1,182 @@
+}
+
+function generateSeats() {
+    const seats = {}
+    const rowStartChar = 'A'.charCodeAt(0)
+    const rowEndChar = lastRow.charCodeAt(0)
+    const colStartChar = '1'.charCodeAt(0)
+    const colEndChar = lastCol.charCodeAt(0)
+
+    for (let row = rowStartChar; row <= rowEndChar; row += 1) {
+        for (let col = colStartChar; col <= colEndChar; col += 1) {
+            const seatNo = `${String.fromCharCode(row)}${String.fromCharCode(col)}`
+            seats[seatNo] = {
+                status: bookingStatus.available,
+                reservedExpiredTime: null,
+            }
+        }
+    }
+    return seats
+}
+
+function reserveSeat(seatNo) {
+    seats[seatNo].status = bookingStatus.reserved
+    seats[seatNo].reservedExpiredTime = Number(new Date()) + (confirmTimeLimit * 1000) // store in timestamp format
+}
+
+function cancelSeat(seatNo) {
+    seats[seatNo] = {
+        status: bookingStatus.available,
+        reservedExpiredTime: null,
+    }
+}
+
+function confirmSeat(seatNo) {
+    delete seats[seatNo]
+
+    booking.push(seatNo)
+
+    if (Object.keys(seats).length === 0) {
+        // start new round
+        round += 1
+        init()
+    }
+}
+
+module.exports.init = () => (init())
+module.exports.dispose = () => {
+    bookings = []
+    seats = null
+}
+
+module.exports.getRemaining = (noOfSeats) => {
+    const _seats = []
+    const now = Number(new Date())
+    const seatNos = Object.keys(seats)
+    if (isNaN(Number(noOfSeats))) noOfSeats = seatNos.length
+    for (let i = 0; i < seatNos.length; i += 1) {
+        const seat = seats[seatNos[i]]
+        if (seat.status === bookingStatus.available ||
+            (
+                seat.status === bookingStatus.reserved &&
+                now >= seat.reservedExpiredTime
+            )
+        ) {
+            _seats.push(seatNos[i])
+        }
+        if (_seats.length >= noOfSeats) {
+            break
+        }
+    }
+    return _seats
+}
+
+module.exports.reserve = (seatNo) => {
+    const now = Number(new Date())
+    if (
+        seatNo in seats && 
+        seats[seatNo].status === bookingStatus.available
+    ) {
+        // seat valid and available
+    } else if (
+        seatNo in seats && 
+        seats[seatNo].status === bookingStatus.reserved &&
+        now >= seats[seatNo].reservedExpiredTime
+    ) {
+        // reserve expire
+    } else {
+        // seatNo not valid or not available
+        // let get an available seat for user
+        const seatNos = Object.keys(seats)
+        for (let i = 0; i < seatNos.length; i += 1) {
+            const seat = seats[seatNos[i]]
+            if (seat.status === bookingStatus.available || 
+                (
+                    seat.status === bookingStatus.reserved &&
+                    now >= seat.reservedExpiredTime
+                )
+            ) {
+                seatNo = seatNos[i]
+                break
+            }
+        }
+    }
+    if (!(seatNo in seats)) {
```

Comment:
Should we go find the seat next round if there is no seat available in this round?
It seems like he/she need to confirm the last seat and then we can go further to next round.

We should have rules for this, if last seat isn't confirmed yet and customers are coming. they can/can't get next round seat. @varokas

Reply:
which is fine. we want to have some contention going on. but in light of this (https://github.com/bigbearsio/strength/issues/6#issuecomment-391597105) we probably will just remove rounds altogether. 

Let's skip this concern and merge this after other comments are addressed. Then we can do another PR to remove rounds?